gdal/gcore/gdal_misc.cpp

```diff
@@ -4070,6 +4071,43 @@ int GDALCanFileAcceptSidecarFile(const char* pszFilename)
     return TRUE;
 }
 
+/************************************************************************/
+/*                   GDALCanReliablyUseSiblingFileList()                */
+/************************************************************************/
+
+/* Try to address https://github.com/OSGeo/gdal/issues/2903 */
+/* On Apple HFS+ filesystem, filenames are stored in a variant of UTF-8 NFD */
+/* (normalization form decomposed). The filesystem takes care of converting */
+/* precomposed form as often coming from user interface to this NFD variant */
+/* See https://stackoverflow.com/questions/6153345/different-utf8-encoding-in-filenames-os-x */
+/* And readdir() will return such NFD variant encoding. Consequently comparing */
+/* the user filename with ones with readdir() is not reliable */
```

Comment:
HFS+ has been replaced by Apple File System (APFS). 

> "APFS accepts only valid UTF-8 encoded filenames for creation, and preserves both case and normalization of the filename on disk in all variants. In macOS High Sierra, APFS is normalization-insensitive in both the case-insensitive and case-sensitive variants, using a hash-based native normalization scheme." 



> "APFS preserves the normalization of the filename and uses hashes of the normalized form of the filename to provide normalization insensitivity"

https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/APFS_Guide/FAQ/FAQ.html

Reply:
@dg0yt Hum, so with APFS, if one always use precomposed form, my understanding is that readdir() should also return it. So I suspect the issue might be that one of the .tif or .tfw file (or both) would have been created using one of the UTF-8 encoding (likely the decomposed one if using MacOS specific API), but the string passed to GDAL for opening would be with another one (likely the precomposed one)